viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts

```diff
@@ -59,6 +60,40 @@ export class Cdf360ImageEventProvider implements Image360Provider<Metadata> {
     return faces;
   }
 
+  private async listEvents(filter: EventFilter): Promise<CogniteEvent[]> {
+    const partitions = 10;
+    return (
+      await Promise.all(
+        range(1, partitions + 1).map(async index => {
+          const result = () => this._client.events.list({ filter, limit: 1000, partition: `${index}/${partitions}` });
+          const pageGenerator = this.loadCursorToCompletion(result);
+          const pages = await this.combinePages(pageGenerator);
+          return pages.flat();
```

Comment:
I think you can write something like
```
const list = this._client.events.list(...).autoPagingToArray({ limit: Infinity });
```
We use this in `viewer/packages/data-providers/src/pointcloud-stylable-object-providers/CdfPointCloudStylableObjectProvider.ts`.